lib/matplotlib/dates.pyi

```diff
@@ -0,0 +1,20 @@
+import datetime
+from collections.abc import Sequence
+import numpy as np
+import numpy.typing as npt
+
+TZ = str | datetime.tzinfo
+DateNum = float | Sequence[float]
+
+def _get_tzinfo(tz: TZ | None) -> datetime.tzinfo: ...
+def _reset_epoch_test_example() -> None: ...
+def set_epoch(epoch: str) -> None: ...
+def get_epoch() -> str: ...
+def _dt64_to_ordinalf(d: np.datetime64 | npt.NDArray[np.datetime64]) -> npt.NDArray[np.floating]: ...
+def _from_ordinalf(x: float, tz: TZ | None) -> datetime.datetime: ...
+def datestr2num(d: str | Sequence[str], default: datetime.datetime | None) -> DateNum: ...
+def date2num(d: datetime.datetime | np.datetime64 | Sequence[datetime.datetime] | Sequence[np.datetime64]) -> DateNum: ...
```

Comment:
Can we overload, so that the dimensionality conservation (scalar vs. sequence) is captured? Also, the sequence output is actually always an array, so we can be more precise here.
```suggestion
@overload
def date2num(d: datetime.datetime | np.datetime64) -> float: ...
@overload
def date2num(d: Sequence[datetime.datetime] | Sequence[np.datetime64]) -> npt.NDArray[np.floating]: ...
```
Possibly also for `datestr2num` and `num2timedelta`.

Reply:
Added overloads